monitor/xfeemngr/tokenprice/buffer.go

```diff
@@ -4,34 +4,45 @@ import (
 	"context"
 	"sync"
 
-	"github.com/omni-network/omni/lib/errors"
 	"github.com/omni-network/omni/lib/log"
 	"github.com/omni-network/omni/lib/tokens"
 )
 
 type Buffer struct {
-	mu     sync.RWMutex
-	buffer map[tokens.Token]float64 // map token to price
-	pricer tokens.Pricer
-	opts   *Opts
+	mu        sync.RWMutex
+	streaming bool
+	buffer    map[tokens.Token]float64 // map token to price
+	pricer    tokens.Pricer
+	tokens    []tokens.Token
+	opts      *Opts
 }
 
 // NewBuffer creates a new token price buffer.
 //
 // A token price buffer maintains a buffered view of token prices for multiple
 // tokens. Buffered token prices are not updated unless they are outside the
 // threshold percentage. Start steaming token prices with Buffer.Stream(ctx).
-func NewBuffer(price tokens.Pricer, opts ...func(*Opts)) (*Buffer, error) {
-	o, err := makeOpts(opts)
-	if err != nil {
-		return nil, errors.Wrap(err, "invalid options")
+func NewBuffer(price tokens.Pricer, optOrTokens ...any) *Buffer {
+	o := defaultOpts()
+	tkns := make([]tokens.Token, 0)
+
+	for _, optOrToken := range optOrTokens {
+		if opt, ok := optOrToken.(func(*Opts)); ok {
+			opt(o)
+		}
+
+		if token, ok := optOrToken.(tokens.Token); ok {
+			tkns = append(tkns, token)
+		}
```

Comment:
This may be unecessary, but I like being able to pass in tokens expliclty. i.e.

```golang
tokenprice.NewBuffer(pricer, tokens.ETH, tokens.OMNI)
```